### PR TITLE
add integration test for operator revision and update uninstall output

### PR DESCRIPTION
**Pull request by richardwxn**
1. add integration test for operator revision: for https://github.com/istio/istio/issues/23479
2. update uninstall output, for https://github.com/istio/istio/issues/24360
- for the case of more than 30 proxies still pointing to the old control plane, just output the count instead of listing all of them
- truncate the to be removed resource list, change to group by kind. For example: kind1:name1, kind1: name2, kind2:name2....change to kind1: name1, name2..., kind2: name3,...

### CODE REVIEW

**Comment by richardwxn** [comment]
@howardjohn @ostromart Can you PTAL?

**Comment by richardwxn** [comment]
@esnible Can you review the part of the UX change, trying to make the output cleaner when running `istioctl uninstall`, details can be found in the description section

File: tests/integration/operator/switch_cr_test.go [hunk 1]
```diff
@@ -113,19 +105,21 @@ func TestController(t *testing.T) {
...
+				"operator", "init",
+				"--hub=" + s.Hub,
+				"--tag=" + s.Tag,
+				"--manifests=" + ManifestPath,
+				"--revision=" + CanaryRevisionName,
+			}
+			// install second operator deployment with different revision
+			istioCtl.InvokeOrFail(t, initCmd)
+
+			verifyInstallation(t, ctx, istioCtl, "default", CanaryRevisionName, cs)
```
Comment (howardjohn) [inline tests/integration/operator/switch_cr_test.go:121]: where does the IOP CR come from?

**Review by howardjohn — APPROVED** [review]
LGTM, confused how it works though

**Review by esnible — APPROVED** [review]

**Pull request merged**